lib/realtime_web/channels/realtime_channel/logging.ex

```diff
@@ -43,13 +43,12 @@ defmodule RealtimeWeb.RealtimeChannel.Logging do
   @spec log_error_with_token_metadata(
           code :: binary(),
           msg :: binary(),
-          token :: Joken.bearer_token(),
-          metadata :: keyword()
+          socket :: Phoenix.Socket.t()
         ) :: {:error, %{reason: binary()}}
-  def log_error_with_token_metadata(code, msg, token, metadata \\ []) do
-    if metadata == [], do: Logger.metadata()
-    metadata = update_metadata_with_token_claims(metadata, token)
-    log_error_message(:error, code, msg, metadata)
+  def log_error_with_token_metadata(code, msg, %{assigns: %{access_token: access_token, tenant: tenant_id}} = socket) do
+    Logger.metadata(external_id: tenant_id, project: tenant_id)
```

Comment:
`Logger.metadata` AFAIK updates the process dictionary to do this. Do we really need to do this at this stage?

It feels like it's enough to do this at the `RealtimeChannel.join` level as it will be set for the process from the point onwards. No need to set it for every log statement